[Lim Wei Jun] iP

limweijun opened the pull request:
**Duke**
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))
DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :grinning:

And it is **FREE**!

Features:
:ballot_box_with_check: Managing tasks
:white_large_square: Managing deadlines (coming soon)
:white_large_square: Reminders (coming soon)

---
If you Java programmer, you can use it to practice Java too. Here's the main method:
``` Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

limweijun committed:
Completed Level-1

limweijun committed:
Completed Level-2

limweijun committed:
Completed Level-3

limweijun committed:
Completed Level-4

limweijun committed:
Completed TextUITesting

limweijun committed:
Completed Level-5

limweijun committed:
Completed Level-6

limweijun committed:
Completed Level-6

limweijun committed:
Completed A-TextUiTesting

limweijun committed:
Completed Level-6

limweijun committed:
Completed Level-7

limweijun committed:
Completed Level 8

limweijun committed:
Resolved conflicts

limweijun committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

limweijun committed:
Completed A-More-OOP

limweijun committed:
Completed A-Packages

limweijun committed:
Completed A-Gradle

limweijun committed:
Completed A-JUnit

limweijun committed:
branch-A-JavaDoc

limweijun committed:
COmpleted branch-A-JavaDoc

limweijun committed:
Completed branch-A-CodingStandard

limweijun committed:
Completed A-Coding-Standard

limweijun committed:
Completed Branch-Level-9

limweijun committed:
Resolved conflicts with A-CodingStandard

limweijun committed:
Merge branch 'branch-level-9'

limweijun committed:
Updated changes to java docs and Storage class

limweijun committed:
Made Changes to Java Docs and Test

limweijun committed:
resolved error

limweijun committed:
Resolved Merging Conflict with A-CodingStandard

limweijun committed:
Changed mainClassName in build.gradle

limweijun committed:
Added Duke.class in Duke Package

yeojunjie commented on src/test/java/Duke/TodoTest.java:
```diff
@@ -0,0 +1,12 @@
+public class TodoTest {
+
+    @Test
+    public void todoTest() {
```
Hello Wei Jun, I would recommend following the conventions:
1. The name of each test method should follow the following format:
`featureUnderTest_testScenario_expectedBehavior()`
2. Strive towards having one `assert` statement in each test method.

yeojunjie commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+                String type = details[0];
+                String[] date;
+                switch (type) {
+                    case "T":
```
Hello Wei Jun,
In CS2103T, we need to ensure that `switch` and its `case`s be at the same level of indentation.

yeojunjie commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,100 @@
+        System.out.println("Hi I'm Duke.Duke, What can I do for you?");
+    }
+
+    /**
```
Hi Wei Jun, the JavaDocs comments could be more descriptive.
I would recommend changing 
```
/**
 * The method is a static and takes in two parameter
 * @param task of type Task
 * @param size of type Int
 */
```
to
```
/**
 * Prints to System.out to inform the user that a new task has been added, and how many tasks there are now.
 * @param the Task that was just added
 * @param size the new number of Tasks
 */
```

yeojunjie commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+ * The class is an abstract class with parameters task and done
+ * @author LimWeiJun
+ */
+abstract public class Task {
```
Hi Wei Jun,
I like how the fields and methods are named. It is easy to tell that `isDone` is a boolean just from its name. The method names also all start with verbs.
However, the name `task` could be more descriptive. May I suggest `description`?

yeojunjie reviewed (changes requested):
Hello, I have left some comments regarding coding standards and conventions.

notle1706 commented on src/main/java/Duke/Event.java:
```diff
@@ -0,0 +1,49 @@
+ * @author LimWeiJun
+ */
+public class Event extends Task {
+    LocalDateTime at;
```
Perhaps you could add a private modifier here?

notle1706 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,74 @@
+                try {
+                    String[] details;
+                    Task task;
+                    switch (strs[0]) {
```
CS2103t coding standards requires switch and case to be on the same level of indentation.

notle1706 commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+     */
+    public static boolean updateFile(Task task) {
+        try {
+            String taskStr = task.getType() == 'T' ?
```
Perhaps you could put the ternary operator `?` on the next line?
CS2103t conventions requires a break before "_operator-like_" symbols.

notle1706 commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+     * @param tasks The input to be received
+     * @return returns a boolean
+     */
+    public static boolean rewriteEntireFile(ArrayList<Task> tasks) {
```
Perhaps you could try renaming `rewriteEntireFile` as CS2103T requires boolean methods to sound like booleans.
>Boolean variables/methods should be named to sound like booleans

notle1706 reviewed (commented):
LGTM. Just some nits to fix.

limweijun committed:
Completed Level-10

limweijun committed:
Completed Level 10.

limweijun committed:
Completed Branch Level 10

limweijun committed:
Completed Level 10

limweijun committed:
Completed Level 10

limweijun committed:
Completed FXML

limweijun committed:
Assertions: handle invalid user input

Users may enter invalid commands such as delete 5 when the length of the
task is only 3 for e.g.

Assertions have been added to line 24, 26, 27, 34, 36, and 37 to check
if user input is a valid integer, > 0 and <= length of task list

For more information on assertions, refer to:
https://nus-cs2103-ay2223s1.github.io/website/schedule/week5/topics.html

limweijun committed:
Code Quality: Reduce arrowheads and nested methods

Improves code readability by extracting nested methods into
smaller ones in Parser.java and Storage.java.

Improves code readability by removing complicated expressions with
extractions.

Improves code readability by removing unused parameters to avoid
confusion to users.

For more information on code readability, refer to:
https://nus-cs2103-ay2223s1.github.io/website/schedule/week5/topics.html

limweijun committed:
Edited assert statements

limweijun committed:
Completed branch-B-Snooze

limweijun committed:
Fixed assertions statements

limweijun committed:
Merge pull request #2 from limweijun/branch-A-Assertions

Branch a assertions

limweijun committed:
Merge branch 'master' into branch-A-CodeQuality

limweijun committed:
Merge pull request #3 from limweijun/branch-A-CodeQuality

Code Quality: Reduce arrowheads and nested methods

limweijun committed:
resolved merging conflicts with B-Snooze

limweijun committed:
Updated Assertion changes to Parser

limweijun committed:
Added Ui.png in docs folder

limweijun committed:
Updated README.md

limweijun committed:
Updated README.md

limweijun committed:
Replace duke and user image

limweijun committed:
Update UserGuide
